Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/LeaderBoard.hs

```diff
@@ -125,17 +130,20 @@ getWeeklyDriverLeaderBoard (personId, merchantId, merchantOpCityId) fromDate toD
   driversWithScoresMap :: [(Text, Double)] <- concat <$> Redis.withNonCriticalRedis (Redis.get $ RideEndInt.makeCachedWeeklyDriverLeaderBoardKey merchantId fromDate toDate)
   let driverIds = map (Id . fst) driversWithScoresMap
   driverNamesMap <- HM.fromList . map (\driver -> (driver.id.getId, getPersonFullName driver)) <$> QPerson.getDriversByIdIn driverIds
+  driverGender <- HM.fromList . map (\driver -> (driver.id.getId, Just driver.gender)) <$> QPerson.getDriversByIdIn driverIds
```

Comment:
We already have queried driver table in above line why to query again. You can populate the data from there itself.